ignite/contrib/handlers/lr_finder.py

```diff
@@ -241,6 +244,16 @@ def plot(
             lrs = lrs[skip_start:-skip_end]
             losses = losses[skip_start:-skip_end]
 
+        # Check to show the suggested learning rate
+        if display_suggestion:
+            idx = self._min_grad_idx + skip_start - skip_end
```

Comment:
This can be wrong if user does not call `lr_suggestion` before.
I think it would be better to call `lr_suggestion` here instead of using `self._min_grad_idx` which we should not store as instance attribute.

Reply:
Okay, plot the values right we have to find the index `sug_lr` not in the history but in the `lrs` list in the `plot` function example: 
```
        lrs = self._history["lr"]
        losses = self._history["loss"]
        if skip_end == 0:
            lrs = lrs[skip_start:]
            losses = losses[skip_start:]
        else:
            lrs = lrs[skip_start:-skip_end]
            losses = losses[skip_start:-skip_end]

        if display_suggestion:
            sug_lr = self.lr_suggestion()
            idx = lrs.index(sug_lr)
            print(idx)
            equivalent_loss = losses[int(idx)]
```
And that can go wrong and raise error like 
```
 # Check to show the suggested learning rate
        if display_suggestion:
            sug_lr = self.lr_suggestion()
            idx = lrs.index(sug_lr)
            ValueError: 0.00012589254117941674 is not in list
```
To avoid this we can find the index of the `sug_lr` in the list but this also can go wrong and the plot can be something like this
![dummy](https://user-images.githubusercontent.com/54319724/115875573-a4d42f80-a445-11eb-951b-4634029ed143.jpg)

Reply:
The simplest solution is to manipulate the `skip_start`, for example we can put it initially to zero instead of 10 in the plot function and mention that in the docs, or even remove it!

Reply:
@KickItLikeShika I do not understand how skip_start can solve the issues. For me the simplest solution without changing the API is to draw vertical line instead of a circle if we can not determine correctly the loss value associated with suggested lr. Please, be more explicit in your suggestions...

Reply:
The problem isn't with finding the index of the loss, It's with finding the index of the suggested learning rate in the `lrs` list in the plot function, drawing a vertical line won't change anything in this case, and If we tried to find the suggested learning rate in the `history` the plot can be something like the image above

Reply:
Let's assume that `skip_start = 0`, almost the `self._history["lrs"]` will be equal to `lrs` list in the plot function (the `skip_end` won't matter as we mostly ignore that part), then we will have no issues finding the suggested learning rate, and that will get us a proper plot at the end

Reply:
I will call `lr_suggestion` then I got the value of `suggested_lr` that found in the whole history, with its index in the history I will find the corresponding loss in the history, then I can't apply the `skip_start` and `skip_end`, right?

Reply:
For more details:

If indexes found in the history:
```
the history of lrs: [1, 10]
the history of losses: [1, 10]
lrs list in plot function after skip star and end: [3, 9]
losses list in plot function after skip start and ed: [3, 9]
suggested_lr: 2
corresponding loss: 2
```
-------> the plot is wrong

If indexes found the lists
```
Raise a ValueError the lr is not found
```

Reply:
@KickItLikeShika please try that instead of speculating. I'm sure that it is not a big deal to make it work with out `self._min_grad_idx`. If there is a real blocker, put explicitly the code line and give an explanation why you think we can not go this way.

Reply:
I'm already trying everything
Here is the code you told me to try (I think we can't apply `skip_start` and `skip_end` as we have the values not the indexes)
```
if display_suggestion:
            sug_lr = self.lr_suggestion()
            idx = self._history["lr"].index(sug_lr)
            corresponding_loss = self._history["loss"][int(idx)]
            fig, ax = plt.subplots()
            ax.scatter(
                sug_lr, corresponding_loss, s=75, marker="o", color="red", zorder=3,
            )
```
And here is the result
![dummy](https://user-images.githubusercontent.com/54319724/115885558-5c6e3f00-a450-11eb-89bb-667b9d93d2b9.jpg)

Reply:
OK, have you tried to move this code
```python
        if skip_end == 0:
            lrs = lrs[skip_start:]
            losses = losses[skip_start:]
        else:
            lrs = lrs[skip_start:-skip_end]
            losses = losses[skip_start:-skip_end]
```
after `if display_suggestion: ...` code ??

Reply:
I tried it now, and it gives the same results

Reply:
OK, please find another solution without `self._min_grad_idx`.

Reply:
The result with `skip_start = 0` (always gives correct results)

![dummy](https://user-images.githubusercontent.com/54319724/115887814-98a29f00-a452-11eb-910c-a402a2802435.jpg)

To wrap up: The issue is related to the visualization interval, what happens is: if the suggested lr index is less than the `skip_start` value the red dot won't be on the curve, It will be before the curve as I mentioned above (assuming we will keep getting the index from the history not the lists)

Suggested solution:
- Make the `skip_start` initially to zero, and put a check inside the `if display_suggestion:` to check if the index of the lr is smaller than the `skip_start`, then we can raise a `ValueError` to ask to reduce the `skip_start` value, what do you think? I can think about another solution...